packages/core/src/generatePolicy.js

```diff
@@ -126,7 +170,7 @@ function createModuleInspector(opts = {}) {
         allowReturnOutsideFunction: true,
         errorRecovery: true,
       })
-    if (includeDebugInfo && ast.errors && ast.errors.length) {
+    if (includeDebugInfo && isParsedAST(ast) && ast.errors.length) {
```

Comment:
This is clever. And I don't even mind the runtime overhead vs clarity in this case.

Reply:
`ast.errors` doesn't mean very much to TS' type inference, but a proper type guard like `isParsedAST()` does.